### PR TITLE
fix(API): Do not use base64 for text types

### LINKED ISSUE #853: BUG: TableReport in skore UI
### Describe the bug

Display of `CramÃ©r's V` and `â` (and some other stuff like that) of a skrub TableReport in the skore UI:

Capture d’écran 2024-12-03 à 10 30 12

### Steps/Code to Reproduce

```python
from skrub.datasets import fetch_employee_salaries
from skrub import TableReport

dataset = fetch_employee_salaries()
employees_df, salaries = dataset.X, dataset.y
my_tablereport = TableReport(employees_df)
my_project.put("my_tablereport", my_tablereport)
my_tablereport
```

### Environment

```shell
System:
    python: 3.12.4 | packaged by Anaconda, Inc. | (main, Jun 18 2024, 10:07:17) [Clang 14.0.6 ]
executable: /Users/sylvaincom/Documents/GitHub/skore/.venv/bin/python
   machine: macOS-15.1.1-arm64-arm-64bit

Python dependencies:
        skore: 0.0.0+unknown
          pip: 24.3.1
   setuptools: None
    diskcache: 5.6.3
      fastapi: 0.115.3
 plotly<6,>=5: None
      pyarrow: 18.1.0
         rich: 13.9.3
        skops: 0.10.0
      uvicorn: 0.32.0
```

### PR DESCRIPTION
**Pull request by rouk1**
Fixes #853.
Screenshot 2024-12-03 at 13 56 30

Bonus: projects_routes.py coverage is now higher.

### CODE REVIEW

File: skore/tests/integration/ui/test_ui.py [hunk 1]
```diff
@@ -98,3 +102,42 @@ def test_serialize_polars_series_with_missing_values(client, in_memory_project):
...
+    np_array = numpy.array([1, 2, 3, 4])
+    in_memory_project.put("np array", np_array)
+
+    response = client.get("/api/project/items")
+    assert response.status_code == 200
+    project = response.json()
+    assert len(project["items"]["np array"][0]["value"]) == 4
+
+
+def test_serilialize_sklearn_estimator(client, in_memory_project):
```
Comment (augustebaum) [inline skore/tests/integration/ui/test_ui.py:117]:
```suggestion
def test_serialize_sklearn_estimator(client, in_memory_project):
```

**Review by augustebaum — APPROVED** [review]
Works for me

File: skore/src/skore/ui/project_routes.py [hunk 2]
```diff
@@ -78,8 +78,12 @@ def pandas_dataframe_to_serializable(df: pandas.DataFrame):
                 value = item.estimator_html_repr
                 media_type = "application/vnd.sklearn.estimator+html"
             elif isinstance(item, MediaItem):
-                value = base64.b64encode(item.media_bytes).decode()
-                media_type = f"{item.media_type};base64"
+                if "text" in item.media_type:
+                    value = item.media_bytes.decode()
```
Comment (thomass-dev) [inline skore/src/skore/ui/project_routes.py:82]:
```suggestion
                    value = item.media_bytes.decode(item.media_encoding)
```

**Comment by sylvaincom** [comment]
Thanks!

**Pull request merged**